pandas/tests/io/formats/test_format.py

```diff
@@ -3240,14 +3240,29 @@ def test_repr_html_ipython_config(ip):
 
 
 @pytest.mark.parametrize("method", ["to_string", "to_html", "to_latex"])
+@pytest.mark.parametrize("encoding", [None, "utf-8", "gbk", "foo"])
 def test_filepath_or_buffer_arg(
-    float_frame, method, filepath_or_buffer, assert_filepath_or_buffer_equals
+    float_frame,
+    method,
+    filepath_or_buffer,
+    assert_filepath_or_buffer_equals,
+    encoding,
+    filepath_or_buffer_id,
 ):
     df = float_frame
-    expected = getattr(df, method)()
 
-    getattr(df, method)(buf=filepath_or_buffer)
-    assert_filepath_or_buffer_equals(expected)
+    if filepath_or_buffer_id not in ["string", "pathlike"] and encoding is not None:
+        with pytest.raises(
+            ValueError, match="buf is not a file name and encoding is specified."
+        ):
+            getattr(df, method)(buf=filepath_or_buffer, encoding=encoding)
+    elif encoding == "foo":
```

Comment:
I think you can remove the invalid encoding; this doesn't test any function pandas provides rather just builtin Python functionality

Reply:
@simonjayhawkins thoughts?

Reply:
The reason I asked for it to be added was so that the precedence of the Exceptions was checked and to confirm the encoding parameter was passed to the builtin function.

Reply:
Oh OK my mistake - just didn't see that was asked for previously (lost in GH comments)

Reply:
so agree with https://github.com/pandas-dev/pandas/pull/28951#discussion_r337124564 but this sort of compensates. If we conform the encoding is passed, then reading back in is only testing Python functionality.

float_frame will probably work with any encoding, so maybe best to modify float_frame if encoding=="gbq".

Reply:
this should work...

```diff
diff --git a/pandas/tests/io/formats/test_format.py b/pandas/tests/io/formats/test_format.py
index 096fc6cb4..490cecb41 100644
--- a/pandas/tests/io/formats/test_format.py
+++ b/pandas/tests/io/formats/test_format.py
@@ -73,17 +73,19 @@ def filepath_or_buffer(filepath_or_buffer_id, tmp_path):
 
 
 @pytest.fixture
-def assert_filepath_or_buffer_equals(filepath_or_buffer, filepath_or_buffer_id):
+def assert_filepath_or_buffer_equals(
+    filepath_or_buffer, filepath_or_buffer_id, encoding
+):
     """
     Assertion helper for checking filepath_or_buffer.
     """
 
     def _assert_filepath_or_buffer_equals(expected):
         if filepath_or_buffer_id == "string":
-            with open(filepath_or_buffer) as f:
+            with open(filepath_or_buffer, encoding=encoding) as f:
                 result = f.read()
         elif filepath_or_buffer_id == "pathlike":
-            result = filepath_or_buffer.read_text()
+            result = filepath_or_buffer.read_text(encoding=encoding)
         elif filepath_or_buffer_id == "buffer":
             result = filepath_or_buffer.getvalue()
         assert result == expected
@@ -3250,6 +3252,8 @@ def test_filepath_or_buffer_arg(
     filepath_or_buffer_id,
 ):
     df = float_frame
+    if encoding == "gbk":
+        float_frame.iloc[0, 0] = "造成输出中文显示乱码"
 
     if filepath_or_buffer_id not in ["string", "pathlike"] and encoding is not None:
         with pytest.raises(
```